From akpm@linux-foundation.org  Fri Aug  7 16:57:48 2009
From: Roel Kluin <roel.kluin@gmail.com>
Date: Thu, 06 Aug 2009 16:09:51 -0700
Subject: USB: gadget: Read buffer overflow
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org, akpm@linux-foundation.org, roel.kluin@gmail.com, david-b@pacbell.net
Message-ID: <200908062309.n76N9q3C012318@imap1.linux-foundation.org>


From: Roel Kluin <roel.kluin@gmail.com>

Check whether index is within bounds before testing the element.

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/usb/gadget/composite.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -602,7 +602,7 @@ static int get_string(struct usb_composi
 			}
 		}
 
-		for (len = 0; s->wData[len] && len <= 126; len++)
+		for (len = 0; len <= 126 && s->wData[len]; len++)
 			continue;
 		if (!len)
 			return -EINVAL;
